packages/vue-pdf-viewer/src/components/PDFViewer.vue

```diff
@@ -1,223 +1,173 @@
-        },
-        pageCount() {
-            return (this.pdfDocument) ? this.pdfDocument.numPages : 0
-        },
-        showPageFitButton() {
-            return this.currentScale !== 'page-fit'
+    source: {
+        required: true,
+        validator: (value) => {
+            return !!value && (typeof value === 'string' || value instanceof Uint8Array)
         }
     },
-    watch: {
-        currentScale(newVal, oldVal) {
-            if (oldVal) {
-                this.pdfViewer.currentScaleValue = this.currentScale
-                const event = new CustomEvent('PDFViewer:scaleChange', { detail: this.pdfViewer.currentScale })
-                window.dispatchEvent(event)
-            }
+    pdfjsViewerOptions: {
+        type: Object,
+        default: function() {
+            return {}
         }
     },
-    methods: {
-        pageFit() {
-            this.pdfViewer.currentScaleValue = this.currentScale = 'page-fit'
+    defaultScale: {
+        required: false,
+        validator: (value) => {
+            if (typeof value === 'string') {
+                return !!value && ['auto', 'page-actual', 'page-fit', 'page-width'].includes(value)
+            } else {
+                return !!value && typeof value === 'number'
+            }
         },
-        decreaseScale() {
-            let newScale = this.pdfViewer.currentScale
-            newScale -= 0.1
-            newScale = newScale.toFixed(2)
-            newScale = Math.floor(newScale * 10) / 10
+        default: 'auto'
+    }
+})
 
-            this.currentScale = (newScale < MIN_SCALE) ? MIN_SCALE : newScale
-        },
-        increaseScale() {
-            let newScale = this.pdfViewer.currentScale
-            newScale += 0.1
-            newScale = newScale.toFixed(2)
-            newScale = Math.ceil(newScale * 10) / 10
+// Events
+const emit = defineEmits(['error'])
 
-            this.currentScale = (newScale > MAX_SCALE) ? MAX_SCALE : newScale
-        },
-        async downloadDocument() {
-            if (!this.allowDocumentDownload) {
-                return
-            }
-            if (this.source instanceof Uint8Array) {
-                this.pdfViewer.downloadManager.downloadData(this.source, this.documentName)
-            } else {
-                if (/^blob:/.test(this.source)) {
-                    const response = await fetch(this.source)
-                    if (response.status === 200) {
-                        const blob = await response.blob()
-                        this.pdfViewer.downloadManager.download(blob, this.source, this.documentName)
-                    } else {
-                        throw new Error('PDFViewer: Fetching blob url failed.')
-                    }
-                } else {
-                    this.pdfViewer.downloadManager.downloadUrl(this.source, this.documentName)
-                }
-            }
+// Reactive data
+const state = ref({
+    pdfViewer: null,
+    pdfDocument: null,
```

Comment:
Is it even necessary that these are reactive? If they would be just normal variables, I guess you wouldn't need `toRaw`.
I think it wouldn't be possible to create a computed based on them anymore, but it would maybe be better to do that over the events anyway.
The `pageCount` computed could be just an updated ref value when the event `pagesloaded` is triggered and I guess there is also an event for "current page changed" or something like that.

Reply:
Good idea, I also thought about removing the reactivity but did not think about using the pdfjs events instead. I changed it now